ToricGeometry: Fix well definedness check; Allow disabling

lkastner opened the pull request.

lkastner committed:
ToricGeometry: Fix well definedness check; Allow disabling

lkastner committed:
ToricGeometry: Refactor toric_morphism constructors

HereAround committed:
[ToricVarieties] Attempt to fix deploy_docs test

HereAround commented:
Thank you @lkastner - generally speaking this looks good to me. And hopefully, this helps to speed up the tests.

When I looked at this PR earlier, of the required tests only the `deploy_docs` tests failed. I tried to fix that for you, but it would seem I made it worse. Still, most of the above failures would seem unrelated to this PR. Maybe this is due to a new julia release of `Singular.jl` or alike.

I suggest to rerun the tests in the light of day and then to reevaluate the situation.

fingolfin commented:
```
┌ Error: no docs found for 'toric_morphism(domain::AbstractNormalToricVariety, mapping_matrix::ZZMatrix)' in `@docs` block in src/AlgebraicGeometry/ToricVarieties/ToricMorphisms.md:26-29
│ ```@docs
│ toric_morphism(domain::AbstractNormalToricVariety, mapping_matrix::ZZMatrix)
│ toric_morphism(domain::AbstractNormalToricVariety, grid_morphism::GrpAbFinGenMap)
│ ```
└ @ Documenter.Expanders ~/.julia/packages/Documenter/bYYzK/src/Utilities/Utilities.jl:32
┌ Error: no docs found for 'toric_morphism(domain::AbstractNormalToricVariety, grid_morphism::GrpAbFinGenMap)' in `@docs` block in src/AlgebraicGeometry/ToricVarieties/ToricMorphisms.md:26-29
│ ```@docs
│ toric_morphism(domain::AbstractNormalToricVariety, mapping_matrix::ZZMatrix)
│ toric_morphism(domain::AbstractNormalToricVariety, grid_morphism::GrpAbFinGenMap)
│ ```
└ @ Documenter.Expanders ~/.julia/packages/Documenter/bYYzK/src/Utilities/Utilities.jl:32
```

lkastner committed:
PolyhedralGeometry: Add transform(Fan, Matrix)

Transform a fan by multiplying with a matrix. The result is not
necessarily a fan, and this can be checked. The check may be omitted.

lkastner committed:
ToricGeometry: Refactor ToricMorphism

- Remove property `image` as this is not necessarily a toric variety.
- Optionaly check whether given map can transform the polyhedral fan of
  `domain` into a proper `codomain`.

lkastner committed:
ToricGeometry: Rename

HereAround reviewed (approved):
Thank you @lkastner for making those nice changes! Very much appreciated.
